app/src/main/java/com/vultisig/wallet/ui/models/TokenSelectionViewModel.kt

```diff
@@ -44,12 +53,61 @@ internal class TokenSelectionViewModel @Inject constructor(
         loadChains()
     }
 
+    @OptIn(ExperimentalStdlibApi::class)
     fun enableToken(coin: Coin) {
-        commitVault(checkNotNull(vaultDb.select(vaultId)) {
-            "No vault with $vaultId"
-        }.let { vault ->
-            vault.copy(coins = vault.coins + coin)
-        })
+        viewModelScope.launch {
+            commitVault(checkNotNull(vaultDb.select(vaultId)) {
+                "No vault with $vaultId"
+            }.let { vault ->
+                when (coin.chain.TssKeysignType) {
+                    TssKeyType.ECDSA -> {
+                        val derivedPublicKey = PublicKeyHelper.getDerivedPublicKey(
+                            vault.pubKeyECDSA,
+                            vault.hexChainCode,
+                            coin.coinType.derivationPath()
+                        )
+                        if (coin.chain == Chain.mayaChain) {
+                            vault.copy(
+                                coins = vault.coins + coin.copy(
+                                    address = MayaChainHelper(
+                                        vault.pubKeyECDSA,
+                                        vault.hexChainCode
+                                    ).getCoin()?.address ?: "",
+                                    hexPublicKey = derivedPublicKey
+                                )
+                            )
+                        } else {
+                            val address = chainAccountAddressRepository.getAddress(
+                                coin.coinType,
+                                PublicKey(
+                                    derivedPublicKey.hexToByteArray(),
+                                    PublicKeyType.SECP256K1
+                                )
+                            )
+                            vault.copy(
+                                coins = vault.coins + coin.copy(
+                                    address = address,
+                                    hexPublicKey = derivedPublicKey
+                                )
+                            )
+                        }
+                    }
+
+                    TssKeyType.EDDSA -> {
+                        val address = chainAccountAddressRepository.getAddress(
+                            coin.coinType,
+                            PublicKey(vault.pubKeyEDDSA.hexToByteArray(), PublicKeyType.ED25519)
+                        )
+                        vault.copy(
+                            coins = vault.coins + coin.copy(
+                                address = address,
+                                hexPublicKey = vault.pubKeyEDDSA
+                            )
+                        )
+                    }
+                }
```

Comment:
Same with TssKeyType, it can probably be supported in a chainAccountAddressRepository::getAddress method, as the difference in the code between the EDDSA and ECDSA is only input parameters (pubKeyEDDSA vs. pubKeyECDSA, PublicKeyType.ED25519 vs. PublicKeyType.SECP256K1)